src/metabase/util.cljc

```diff
@@ -843,3 +843,12 @@
   [xs]
   (or (empty? xs)
       (apply distinct? xs)))
+
+(defn deep-sort-map
+  "Converts a map (with potentially nested maps as values) into a sorted map of sorted maps."
+  [m]
+  (into (sorted-map)
+        (zipmap
+         (keys m)
+         (map (fn [val] (if (map? val) (deep-sort-map val) val))
+              (vals m)))))
```

Comment:
nit: small suggestions - you can just map over the input directly instead of the keys and values separately, and use `cond->` for the conditional. Maybe sth like this?
```suggestion
        (map (fn [[k v]]
               [k (cond-> v (map? v) deep-sort-map)])
             m)))
```

Reply:
I used to use zipkeys to re-write map-vals all the time, is that weird? :) Now that we have `clojure.core/update-vals` I will go ahead and use that instead.